src/session.cpp

```diff
@@ -287,6 +287,45 @@ namespace libtorrent
 	// configurations this will give a link error
 	void TORRENT_CFG() {}
 
+	session_params read_session_params(bdecode_node const& e, std::uint32_t const flags)
```

Comment:
I'm a bit surprised there isn't a function like this already. There's just a member function for doing this now, is there? (which probably should be deprecated).
This makes sense.